packages/camera/camera_platform_interface/lib/src/method_channel/method_channel_camera.dart

```diff
@@ -283,13 +292,17 @@ class MethodChannelCamera extends CameraPlatform {
   @override
   Stream<CameraImageData> onStreamedFrameAvailable(int cameraId,
       {CameraImageStreamOptions? options}) {
+    _installStreamController(onListen: _onFrameStreamListen);
+    return _frameStreamController!.stream;
+  }
+
+  void _installStreamController({Function()? onListen}) {
```

Comment:
In `startVideoRecording` Can we directly pass `streamCallback` into `_installStreamController()` and make it nonNull? A no-op onListen seems a bad state to me.

Reply:
Not sure I totally understand. `streamCallback` is the callback which receives each frame that is processed. `onListen` is different, it's a callback invoked once when the stream is listened on. In the case of `onStreamedFrameAvailable` it needs to call the native `startImageStream` to begin the stream, but on the `startVideoRecording` the camera stream has already been initialised during the call to the native `startVideoRecording`. Therefore, in the case of `startVideoRecording`, `onListen` should be a no-op.

I could make onListen non-null, and have it up to the caller to default it to a no-op function, but that seems to put more burden on the caller. Is that preferable?

Reply:
I meant do we want to have `onListen` for both use cases? It's a bit odd to have one streaming API with onListen, but the other streaming API without.

Reply:
`onListen` isn't part of the public API though. It's an implementation detail inside this class. Nobody using the camera plugin will have to deal with it at all. The only reason that `onListen` is nullable is that, in the case where they want to both stream and record, the initialization is another section of the (internal) code. `StreamController` itself is a class external to the camera plugin and can't be easily modified.

Given that the method is private and internal to the class, is the difference still a concern?